src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsStore.java

```diff
@@ -0,0 +1,134 @@
+package org.jenkinsci.plugins.azurekeyvaultplugin;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.CredentialsStoreAction;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import hudson.model.ModelObject;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import java.util.Collections;
+import java.util.List;
+import jenkins.model.Jenkins;
+import org.acegisecurity.Authentication;
+import org.jenkins.ui.icon.Icon;
+import org.jenkins.ui.icon.IconSet;
+import org.jenkins.ui.icon.IconType;
+import org.kohsuke.stapler.export.ExportedBean;
+
+public class AzureCredentialsStore extends CredentialsStore {
+    private final AzureCredentialsProvider provider;
+    private final AzureCredentialsStoreAction action = new AzureCredentialsStoreAction(this);
+
+    public AzureCredentialsStore(AzureCredentialsProvider provider) {
+        super(AzureCredentialsProvider.class);
+        this.provider = provider;
+    }
+
+    @NonNull
+    @Override
+    public ModelObject getContext() {
+        return Jenkins.getInstance();
+    }
+
+    @Override
+    public boolean hasPermission(@NonNull Authentication authentication, @NonNull Permission permission) {
+        return CredentialsProvider.VIEW.equals(permission)
+                && Jenkins.getInstance().getACL().hasPermission(authentication, permission);
+    }
+
+    @NonNull
+    @Override
+    public List<Credentials> getCredentials(@NonNull Domain domain) {
+        if (Domain.global().equals(domain)
+                && Jenkins.getInstance().hasPermission(CredentialsProvider.VIEW)) {
+            return provider.getCredentials(Credentials.class, Jenkins.getInstance(), ACL.SYSTEM);
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public boolean addCredentials(@NonNull Domain domain, @NonNull Credentials credentials) {
+        throw new UnsupportedOperationException(
+                "Jenkins may not add credentials to Azure Key Vault");
+    }
+
+    @Override
+    public boolean removeCredentials(@NonNull Domain domain, @NonNull Credentials credentials) {
+        throw new UnsupportedOperationException(
+                "Jenkins may not remove credentials to Azure Key Vault");
+    }
+
+    @Override
+    public boolean updateCredentials(@NonNull Domain domain, @NonNull Credentials credentials,
```

Comment:
Any reason for not implementing update?

Reply:
Its purpose is to directly use secrets from key vault. The secrets are not  managed(create/update/delete) here.